core/api/usermanager/v1/usermanager_proto/usermanager.proto

```diff
@@ -0,0 +1,156 @@
+  // This must be set to true if any further operations from this API are meant to succeed.
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_signing_keys = 4;
+}
+
+// BatchCreateUserRequest creates multiple users at once.
+message BatchCreateUserResponse {
+  // users returns the list of created users.
+  repeated keytransparency.v1.proto.User users = 2;
+}
+
+// BatchUpdateUserRequest creates multiple users all at once.
+message BatchUpdateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // users is the set of users to create.
+  repeated keytransparency.v1.proto.User users = 3;
+  // update_mask specifies which fields of user to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 4;
+}
+
+// BatchUpdateUserRequest creates multiple users at once.
+message BatchUpdateUserResponse {
+  // users returns the list of created users.
+  repeated keytransparency.v1.proto.User users = 2;
+}
+
+
+// The UserManagerService holds a set of signing keys which can be used to
+// sign user updates.
+// - This service must be registered with the server in order to submit user
+// create or update requests.
+// - The update requests this service generates will only be valid for users
+// that have added this service's signing keys to their list of authorized_keys.
+// - This service should carefully controll and log access to it's APIs.
+// - This service should rotate its signing keys periodically.
+//   New signing keys should be published far enough in advance that they can be added
+//   by everyone before they are used to avoid service disruption.
+//
+// Resources:
+// - domains/apps/keyset
+//   Each combination of domain_id and app_id has it's own keyset.
+// - domains/apps/users
+//   Users can be created and updated through this endpoint.
+service UserManagerService {
+
+  // GetKeySet returns a list of public keys (a keyset) that corresponds to the signing keys
+  // this service has for a given domain and app.
+  rpc GetKeySet (GetKeySetRequest) returns (keymaster.proto.KeySet) {
+    option (google.api.http) = {
+      get: "/usermanager/v1/domains/{domain_id}/apps/{app_id}/keyset"
+    };
+  }
+
+  // CreateUser creates a new user and initializes it.
+  // If the user already exists, this operation will fail.
+  rpc CreateUser(CreateUserRequest) returns (keytransparency.v1.proto.User) {
+    option (google.api.http) = {
+      post: "/usermanager/v1/domains/{user.domain_id}/apps/{user.app_id}/users/{user.user_id}"
+      body: "user"
+    };
+  }
+
+  // UpdateUserData sets the public key for an user.
+  rpc UpdateUser(UpdateUserRequest) returns (keytransparency.v1.proto.User) {
+    option (google.api.http) = {
+      patch: "/usermanager/v1/domains/{user.domain_id}/apps/{user.app_id}/users/{user.user_id}"
+      body: "user"
+    };
+  }
+
```

Comment:
Do you need to include Get / ListUsers here, or are those covered by another API? If the latter, it may be worth mentioned that somewhere in the docs (ie, "to read users use $otherAPI.$someMethod" or something like it.)